src/net_processing.cpp

```diff
@@ -2936,9 +2936,11 @@ void ProcessMessage(
 
         TxValidationState state;
 
-        nodestate->m_tx_download.m_tx_announced.erase(hash);
-        nodestate->m_tx_download.m_tx_in_flight.erase(hash);
-        EraseTxRequest(hash);
+        for (uint256 hash : {txid, wtxid}) {
+            nodestate->m_tx_download.m_tx_announced.erase(hash);
+            nodestate->m_tx_download.m_tx_in_flight.erase(hash);
```

Comment:
This is a behavior change? We used to call it once on txid OR wtxid, now you calling it twice?

That's probably fine, but why use this for loop at all? `hash` is already defined above, depending on the peer's wtxid support

Reply:
Before, we only ever asked for hashes in one way from a peer, so it was sufficient to only erase that one.

That's no longer the case: outgoing GETDATA requests may be txid or wtxid based, and we don't remember which of the two (it depends on whether it was in response to an INV, or a request for an orphan parent), so either we track that, or just delete both.

It's of course a change in behavior, but that's the point: we're now requesting orphans from wtxidrelay peers (using txid requests), and this is one of the consequences.

Reply:
I see how it works now.
Part of my confusion was that it's part of the "refactor" commit, perhaps better belongs to the next commit.

Reply:
You're right; moved it to the other commit.